sql/pgwire: TestConn failed

[(sql/pgwire).TestConn failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815656&tab=buildLog) on [master@afd27c255065909f357f37f7c38309592e5b0da6](https://github.com/cockroachdb/cockroach/commits/afd27c255065909f357f37f7c38309592e5b0da6):

```
I200318 14:26:20.327552 1297 sql/pgwire/conn_test.go:636  pgx log [info] Dialing PostgreSQL server - map[address:127.0.0.1:42447 network:tcp]
I200318 14:26:20.328301 1297 sql/pgwire/conn_test.go:636  pgx log [info] connection established - map[]
I200318 14:26:20.338485 31 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200318 14:26:20.338744 31 sql/pgwire/conn_test.go:200  advancing Sync
I200318 14:26:20.339089 1297 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=70) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:10.717886ms]
I200318 14:26:20.343557 31 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200318 14:26:20.343637 31 sql/pgwire/conn_test.go:200  advancing Sync
I200318 14:26:20.343803 1297 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:4.315878ms]
I200318 14:26:20.347735 31 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200318 14:26:20.347905 31 sql/pgwire/conn_test.go:200  advancing Sync
I200318 14:26:20.348102 1297 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:4.148565ms]
I200318 14:26:20.353415 31 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200318 14:26:20.353759 31 sql/pgwire/conn_test.go:200  advancing Sync
I200318 14:26:20.353703 1297 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:5.387743ms]
I200318 14:26:20.354072 31 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200318 14:26:20.354215 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:219.943µs]
I200318 14:26:20.354378 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:91.328µs]
I200318 14:26:20.354625 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:50.332µs]
I200318 14:26:20.354807 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:109.818µs]
I200318 14:26:20.355168 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:37.887µs]
I200318 14:26:20.355685 1297 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:435.618µs]
I200318 14:26:21.511716 116 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200318 14:26:21.511897 116 kv/kvserver/stores.go:266  [n1] wrote 0 node addresses to persistent storage
I200318 14:26:25.356363 1297 sql/pgwire/conn_test.go:636  pgx log [info] closed connection - map[]
I200318 14:26:25.356830 31 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.70s)
    conn_test.go:183: read tcp 127.0.0.1:50314->127.0.0.1:42447: i/o timeout

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConn PKG=./pkg/sql/pgwire TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConn.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/pgwire).TestConn failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815862&tab=buildLog) on [master@e1e26af4e20a4e7be3b4b157e6d7c72fbf44c10d](https://github.com/cockroachdb/cockroach/commits/e1e26af4e20a4e7be3b4b157e6d7c72fbf44c10d):

```
I200318 16:04:04.971075 1342 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200318 16:04:04.972234 1344 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.34.169.241:37827 Attrs: Locality: ServerVersion:19.2-15 BuildTag:v20.1.0-beta.2-910-ge1e26af StartedAt:1584547444508434728 LocalityAddress:[] ClusterName: SQLAddress:127.34.169.241:37767} ClusterID:3b1fa78d-b8a9-4f55-90ff-cde865d47029 StartedAt:1584547444508434728 LastUp:1584547444508434728}
I200318 16:04:04.984547 13 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200318 16:04:04.984650 13 sql/pgwire/conn_test.go:200  advancing Sync
I200318 16:04:04.984705 1364 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=70) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:28.174127ms]
I200318 16:04:04.990809 13 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200318 16:04:04.990840 13 sql/pgwire/conn_test.go:200  advancing Sync
I200318 16:04:04.990885 1364 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:5.906899ms]
I200318 16:04:05.004723 13 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200318 16:04:05.004916 13 sql/pgwire/conn_test.go:200  advancing Sync
I200318 16:04:05.005029 1364 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:14.084078ms]
I200318 16:04:05.009857 1364 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:4.719353ms]
I200318 16:04:05.009985 13 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200318 16:04:05.009999 13 sql/pgwire/conn_test.go:200  advancing Sync
I200318 16:04:05.010033 13 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200318 16:04:05.010115 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:201.298µs]
I200318 16:04:05.010224 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:83.031µs]
I200318 16:04:05.010517 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:85.013µs]
I200318 16:04:05.010783 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:213.384µs]
I200318 16:04:05.011227 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:56.455µs]
I200318 16:04:05.011746 1364 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:471.973µs]
I200318 16:04:06.873604 117 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200318 16:04:06.873771 117 kv/kvserver/stores.go:266  [n1] wrote 0 node addresses to persistent storage
I200318 16:04:10.012038 1364 sql/pgwire/conn_test.go:636  pgx log [info] closed connection - map[]
I200318 16:04:10.012141 13 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.63s)
    conn_test.go:183: read tcp 127.0.0.1:38832->127.0.0.1:36789: i/o timeout

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConn PKG=./pkg/sql/pgwire TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConn.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/pgwire).TestConn failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1819397&tab=buildLog) on [master@eb8bd6b256e02a8ca9cd1ee9093fd867038ca81e](https://github.com/cockroachdb/cockroach/commits/eb8bd6b256e02a8ca9cd1ee9093fd867038ca81e):

```
I200320 00:06:21.940878 1194 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.147.131.187:36217 Attrs: Locality: ServerVersion:19.2-15 BuildTag:v20.1.0-beta.2-965-geb8bd6b StartedAt:1584662781509167485 LocalityAddress:[] ClusterName: SQLAddress:127.147.131.187:32897} ClusterID:cd859a8e-0e5f-43d7-8e95-9127e98ad544 StartedAt:1584662781509167485 LastUp:1584662781509167485}
I200320 00:06:21.942443 1192 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200320 00:06:21.965464 1198 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=70) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:26.512653ms]
I200320 00:06:21.965929 29 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200320 00:06:21.966034 29 sql/pgwire/conn_test.go:200  advancing Sync
I200320 00:06:21.978008 1198 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:12.291795ms]
I200320 00:06:21.978500 29 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200320 00:06:21.978585 29 sql/pgwire/conn_test.go:200  advancing Sync
I200320 00:06:21.994525 29 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200320 00:06:21.994555 29 sql/pgwire/conn_test.go:200  advancing Sync
I200320 00:06:21.994664 1198 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:16.568265ms]
I200320 00:06:22.000383 1198 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:5.58075ms]
I200320 00:06:22.000705 29 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200320 00:06:22.000753 29 sql/pgwire/conn_test.go:200  advancing Sync
I200320 00:06:22.000834 29 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200320 00:06:22.000977 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:520.014µs]
I200320 00:06:22.001330 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:302.107µs]
I200320 00:06:22.001711 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:107.701µs]
I200320 00:06:22.002043 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:275.668µs]
I200320 00:06:22.002584 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:60.749µs]
I200320 00:06:22.003143 1198 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:518.557µs]
I200320 00:06:23.050285 111 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200320 00:06:23.050525 111 kv/kvserver/stores.go:266  [n1] wrote 0 node addresses to persistent storage
I200320 00:06:27.003499 1198 sql/pgwire/conn_test.go:636  pgx log [info] closed connection - map[]
I200320 00:06:27.003659 29 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.67s)
    conn_test.go:183: read tcp 127.0.0.1:37092->127.0.0.1:38855: i/o timeout

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConn PKG=./pkg/sql/pgwire TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConn.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/pgwire).TestConn failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1844651&tab=buildLog) on [master@eb5b73f4df83ddacff5c489d431ba476a8d118b4](https://github.com/cockroachdb/cockroach/commits/eb5b73f4df83ddacff5c489d431ba476a8d118b4):

```
I200401 22:50:42.553380 978 sql/pgwire/conn_test.go:636  pgx log [info] connection established - map[]
I200401 22:50:42.578325 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200401 22:50:42.578557 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.578944 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=70) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:25.332829ms]
I200401 22:50:42.583018 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200401 22:50:42.583037 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.583379 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:3.793972ms]
I200401 22:50:42.621656 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200401 22:50:42.621684 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.621710 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:37.216245ms]
I200401 22:50:42.641044 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200401 22:50:42.641070 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.641158 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:19.265598ms]
I200401 22:50:42.641619 23 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200401 22:50:42.641870 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:607.683µs]
I200401 22:50:42.642286 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:287.001µs]
I200401 22:50:42.642886 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:223.721µs]
I200401 22:50:42.649433 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:6.488598ms]
I200401 22:50:42.650173 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:82.482µs]
I200401 22:50:42.650819 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:591.008µs]
I200401 22:50:44.000026 118 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200401 22:50:44.000208 118 kv/kvserver/stores.go:266  [n1] wrote 0 node addresses to persistent storage
I200401 22:50:44.308822 491 kv/kvserver/replica_consistency.go:255  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:1398 LastUpdateNanos:1585781442331769745 IntentAge:0 GCBytesAge:0 LiveBytes:-34064 LiveCount:-672 KeyBytes:-32645 KeyCount:-672 ValBytes:-1419 ValCount:-672 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200401 22:50:47.656983 978 sql/pgwire/conn_test.go:636  pgx log [info] closed connection - map[]
I200401 22:50:47.657096 23 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.49s)
    conn_test.go:183: read tcp 127.0.0.1:59078->127.0.0.1:34679: i/o timeout

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConn PKG=./pkg/sql/pgwire TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConn.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

took a brief look, having trouble reproducing this locally, even with stressrace.

looks like there's a race with the cleanup here. in the example above:

>     conn_test.go:183: read tcp 127.0.0.1:59078->127.0.0.1:34679: i/o timeout

looks like 34679 is the psql server, and 59078 is the client. seems like something with the connection closing yet the server not completely started up is the cause of this to fail.

----

full log below:

```
------- Stdout: -------
=== RUN   TestConn
W200401 22:50:42.239881 23 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200401 22:50:42.258601 23 server/server.go:1082  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200401 22:50:42.284504 23 server/config.go:573  [n?] 1 storage engine initialized
I200401 22:50:42.284545 23 server/config.go:576  [n?] RocksDB cache size: 128 MiB
I200401 22:50:42.284555 23 server/config.go:576  [n?] store 0: in-memory, size 0 B
I200401 22:50:42.284752 23 server/server.go:1543  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200401 22:50:42.284821 23 server/server.go:1553  [n?] **** add additional nodes by specifying --join=127.215.72.226:35641
I200401 22:50:42.295220 23 server/node.go:324  [n?] **** cluster 225a81ef-b25e-4c3a-910d-fe81904e33ea has been created
I200401 22:50:42.295741 23 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.215.72.226:35641" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:16 > build_tag:"v20.1.0-beta.4-309-geb5b73f" started_at:1585781442295464494 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.215.72.226:43259" > 
I200401 22:50:42.300807 23 server/node.go:425  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=16 KiB), ranges=32, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=166.00 pMax=13326.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
W200401 22:50:42.301355 196 kv/kvserver/store.go:1541  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200401 22:50:42.302350 23 kv/kvserver/stores.go:247  [n1] read 0 node addresses from persistent storage
I200401 22:50:42.302464 23 server/node.go:642  [n1] connecting to gossip network to verify cluster ID...
I200401 22:50:42.302505 23 server/node.go:662  [n1] node connected via gossip and verified as part of cluster "225a81ef-b25e-4c3a-910d-fe81904e33ea"
I200401 22:50:42.316987 23 server/node.go:509  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200401 22:50:42.317230 23 server/server.go:1659  [n1] starting http server at 127.215.72.226:33177 (use: 127.215.72.226:33177)
I200401 22:50:42.317321 23 server/server.go:1664  [n1] starting postgres server at 127.215.72.226:43259 (use: 127.215.72.226:43259)
I200401 22:50:42.317374 23 server/server.go:1666  [n1] starting grpc server at 127.215.72.226:35641
I200401 22:50:42.317423 23 server/server.go:1667  [n1] advertising CockroachDB node at 127.215.72.226:35641
I200401 22:50:42.322740 208 sql/temporary_schema.go:451  [n1] running temporary object cleanup background job
I200401 22:50:42.375283 208 sql/temporary_schema.go:485  [n1] found 0 temporary schemas
I200401 22:50:42.376147 208 sql/temporary_schema.go:534  [n1] completed temporary object cleanup job
I200401 22:50:42.376184 208 sql/temporary_schema.go:562  [n1] temporary object cleaner next scheduled to run at 2020-04-01 23:20:42.322636471 +0000 UTC
I200401 22:50:42.393865 336 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200401 22:50:42.458021 383 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-16 User:root}
I200401 22:50:42.479665 705 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:eceffbc8-d929-44f1-9179-3782e3354c3a User:root}
I200401 22:50:42.490969 712 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200401 22:50:42.519026 719 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200401 22:50:42.548424 23 server/server.go:1798  [n1] done ensuring all necessary migrations have run
I200401 22:50:42.548881 23 server/server.go:2028  [n1] serving sql connections
I200401 22:50:42.551218 964 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.215.72.226:35641 Attrs: Locality: ServerVersion:19.2-16 BuildTag:v20.1.0-beta.4-309-geb5b73f StartedAt:1585781442295464494 LocalityAddress:[] ClusterName: SQLAddress:127.215.72.226:43259} ClusterID:225a81ef-b25e-4c3a-910d-fe81904e33ea StartedAt:1585781442295464494 LastUp:1585781442295464494}
I200401 22:50:42.551942 962 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200401 22:50:42.552569 23 sql/pgwire/conn_test.go:84  started listener on 127.0.0.1:34679
I200401 22:50:42.552828 978 sql/pgwire/conn_test.go:636  pgx log [info] Dialing PostgreSQL server - map[address:127.0.0.1:34679 network:tcp]
I200401 22:50:42.553380 978 sql/pgwire/conn_test.go:636  pgx log [info] connection established - map[]
I200401 22:50:42.578325 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200401 22:50:42.578557 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.578944 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=70) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:25.332829ms]
I200401 22:50:42.583018 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200401 22:50:42.583037 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.583379 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:3.793972ms]
I200401 22:50:42.621656 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200401 22:50:42.621684 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.621710 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:37.216245ms]
I200401 22:50:42.641044 23 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200401 22:50:42.641070 23 sql/pgwire/conn_test.go:200  advancing Sync
I200401 22:50:42.641158 978 sql/pgwire/conn_test.go:636  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:19.265598ms]
I200401 22:50:42.641619 23 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200401 22:50:42.641870 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:607.683µs]
I200401 22:50:42.642286 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:287.001µs]
I200401 22:50:42.642886 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:223.721µs]
I200401 22:50:42.649433 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:6.488598ms]
I200401 22:50:42.650173 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:82.482µs]
I200401 22:50:42.650819 978 sql/pgwire/conn_test.go:636  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:591.008µs]
I200401 22:50:44.000026 118 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200401 22:50:44.000208 118 kv/kvserver/stores.go:266  [n1] wrote 0 node addresses to persistent storage
I200401 22:50:44.308822 491 kv/kvserver/replica_consistency.go:255  [n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] triggering stats recomputation to resolve delta of {ContainsEstimates:1398 LastUpdateNanos:1585781442331769745 IntentAge:0 GCBytesAge:0 LiveBytes:-34064 LiveCount:-672 KeyBytes:-32645 KeyCount:-672 ValBytes:-1419 ValCount:-672 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I200401 22:50:47.656983 978 sql/pgwire/conn_test.go:636  pgx log [info] closed connection - map[]
I200401 22:50:47.657096 23 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.49s)
    conn_test.go:183: read tcp 127.0.0.1:59078->127.0.0.1:34679: i/o timeout
```

[(sql/pgwire).TestConn failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1882216&tab=buildLog) on [master@a63a670b1a591be2ed38af7f98c1f85e32546596](https://github.com/cockroachdb/cockroach/commits/a63a670b1a591be2ed38af7f98c1f85e32546596):

```
I200417 05:56:06.369706 980 sqlmigrations/migrations.go:932  [schema-change-job-migration] evaluating tables for creating jobs
I200417 05:56:06.369754 980 sqlmigrations/migrations.go:714  [n1] schema change job migration completed
I200417 05:56:06.377614 37 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, CASE WHEN nsp.nspname IN ('pg_catalog', 'public') THEN t.typname ELSE (nsp.nspname || '.') || t.typname END FROM pg_type AS t LEFT JOIN pg_type AS base_type ON t.typelem = base_type.oid LEFT JOIN pg_class AS base_cls ON base_type.typrelid = base_cls.oid LEFT JOIN pg_namespace AS nsp ON t.typnamespace = nsp.oid WHERE ((t.typtype IN ('b', 'p', 'r', 'e')) AND ((base_type.oid IS NULL) OR (base_type.typtype IN ('b', 'p', 'r'))))
I200417 05:56:06.377640 984 sql/pgwire/conn_test.go:643  pgx log [info] Query - map[args:[] rowCount:%!s(int=74) sql:select t.oid,
	case when nsp.nspname in ('pg_catalog', 'public') then t.typname
		else nsp.nspname||'.'||t.typname
	end
from pg_type t
left join pg_type base_type on t.typelem=base_type.oid
left join pg_class base_cls ON base_type.typrelid = base_cls.oid
left join pg_namespace nsp on t.typnamespace=nsp.oid
where (
	  t.typtype in('b', 'p', 'r', 'e')
	  and (base_type.oid is null or base_type.typtype in('b', 'p', 'r'))
	) time:10.693923ms]
I200417 05:56:06.377648 37 sql/pgwire/conn_test.go:200  advancing Sync
I200417 05:56:06.380751 37 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_type AS base_type ON t.typelem = base_type.oid WHERE (t.typtype = 'b') AND (base_type.typtype = 'e')
I200417 05:56:06.380775 37 sql/pgwire/conn_test.go:200  advancing Sync
I200417 05:56:06.380791 984 sql/pgwire/conn_test.go:643  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
  join pg_type base_type on t.typelem=base_type.oid
where t.typtype = 'b'
  and base_type.typtype = 'e' time:3.011263ms]
I200417 05:56:06.384187 37 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname, t.typbasetype FROM pg_type AS t JOIN pg_type AS base_type ON t.typbasetype = base_type.oid WHERE (t.typtype = 'd') AND (base_type.typtype = 'b')
I200417 05:56:06.384235 37 sql/pgwire/conn_test.go:200  advancing Sync
I200417 05:56:06.384253 984 sql/pgwire/conn_test.go:643  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname, t.typbasetype
from pg_type t
  join pg_type base_type on t.typbasetype=base_type.oid
where t.typtype = 'd'
  and base_type.typtype = 'b' time:3.41678ms]
I200417 05:56:06.387514 37 sql/pgwire/conn_test.go:219  executed query: SELECT t.oid, t.typname FROM pg_type AS t JOIN pg_class AS cls ON t.typrelid = cls.oid WHERE (t.typtype = 'c') AND (cls.relkind = 'c')
I200417 05:56:06.387530 37 sql/pgwire/conn_test.go:200  advancing Sync
I200417 05:56:06.387615 984 sql/pgwire/conn_test.go:643  pgx log [info] Query - map[args:[] rowCount:%!s(int=1) sql:select t.oid, t.typname
from pg_type t
	join pg_class cls on t.typrelid=cls.oid
where t.typtype = 'c'
	and cls.relkind='c' time:3.243769ms]
I200417 05:56:06.387736 37 sql/pgwire/conn_test.go:212  stop wait at query: SELECT 1
I200417 05:56:06.387881 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:select 1 time:223.299µs]
I200417 05:56:06.388010 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:select 2 time:97.402µs]
I200417 05:56:06.388224 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:p1 time:82.924µs]
I200417 05:56:06.388379 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:select 4; select 5; select 6; time:129.464µs]
I200417 05:56:06.388727 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:select 9 time:47.793µs]
I200417 05:56:06.389207 984 sql/pgwire/conn_test.go:643  pgx log [info] Exec - map[args:[] commandTag: sql:bogus statement failing to parse time:443.027µs]
I200417 05:56:07.857773 124 gossip/gossip.go:1538  [n1] node has connected to cluster via gossip
I200417 05:56:07.857882 124 kv/kvserver/stores.go:255  [n1] wrote 0 node addresses to persistent storage
I200417 05:56:11.389522 984 sql/pgwire/conn_test.go:643  pgx log [info] closed connection - map[]
I200417 05:56:11.389637 37 util/stop/stopper.go:539  quiescing
--- FAIL: TestConn (5.15s)
    conn_test.go:183: read tcp 127.0.0.1:50442->127.0.0.1:44211: i/o timeout

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestConn PKG=./pkg/sql/pgwire TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #47464 sql/pgwire: TestConn failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConn.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

Instant repro on gceworker

```
make stress PKG=./pkg/sql/pgwire TESTS=TestConn 2>&1 | tee log
--- FAIL: TestConn (5.28s)
    conn_test.go:183: read tcp 127.0.0.1:43692->127.0.0.1:44709: i/o timeout

188 runs completed, 1 failures, over 17s
context canceled
FAIL
FAIL	github.com/cockroachdb/cockroach/pkg/sql/pgwire	17.428s
```

--- a/pkg/sql/pgwire/conn.go
+++ b/pkg/sql/pgwire/conn.go
@@ -213,6 +213,8 @@ func (c *conn) serveImpl(
 	authOpt authOptions,
 	stopper *stop.Stopper,
 ) {
+	defer func() { _ = c.conn.Close() }()
+
 	ctx = logtags.AddTag(ctx, "user", c.sessionArgs.User)
 
 	inTestWithoutSQL := sqlServer == nil
@@ -221,7 +223,6 @@ func (c *conn) serveImpl(
 		authLogger = sqlServer.GetExecutorConfig().AuthLogger
 		sessionStart := timeutil.Now()
 		defer func() {
-			_ = c.conn.Close()
 			if c.authLogEnabled() {
 				authLogger.Logf(ctx, "session terminated; duration: %s", timeutil.Now().Sub(sessionStart))
 			}

